stats/cloud/collector_test.go

```diff
@@ -585,3 +586,107 @@ func TestCloudCollectorAggregationPeriodZeroNoBlock(t *testing.T) {
 	wg.Wait()
 	require.Equal(t, lib.RunStatusQueued, collector.runStatus)
 }
+
+func TestCloudCollectorRecvIterLIAllIterations(t *testing.T) {
+	t.Parallel()
+	tb := httpmultibin.NewHTTPMultiBin(t)
+	tb.Mux.HandleFunc("/v1/tests", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err := fmt.Fprintf(w, `{
+			"reference_id": "123",
+			"config": {
+				"metricPushInterval": "10ms",
+				"aggregationPeriod": "0ms",
+				"aggregationCalcInterval": "40ms",
+				"aggregationWaitPeriod": "5ms"
```

Comment:
You don't need any of these options here, since aggregation is [disabled by default](https://github.com/loadimpact/k6/blob/9554c6e7e9116351fa330623ac28e3242dacb99c/stats/cloud/config.go#L157) and you're not doing anything aggregation-related in your test anyway. Actually, the whole `config` override probably isn't necessary here, given how the `cancel()` at the end of the test would force the cloud collector to push its metrics, whatever the `metricPushInterval`

Reply:
Thanks, fixed in 5de1b502.